### ISSUE TITLE
Responsive design and window size

### ISSUE THREAD

**Issue by DavidVII** [keathley/wallaby#152]
After bringing in some css my app now has a responsive navigation where a window size narrower than 769 pixels wide will turn into a hamburger menu. Since the default window size in wallaby is 100 x 100 I end up running tests against the responsive layout and can not click on links in the responsive menu since it involves an extra step. This leaves me with two options:

1. Account for extra step
2. resize_window in the current session

I opted for the second option, but I have quite a few tests where I have to keep resizing the window. Is there a way to configure the default window size for all acceptance cases? I'd love to set this option globally so that I don't have to keep resizing across all my tests. I'm happy to help if this option isn't available. I just wanted to double check a feature like this doesn't already exist.

Cheers!

**Comment by keathley** [comment]
I don't think there's a window size option to pass on session creation in webdriver. You will have to set it after creating the session. You could add something like this to your AcceptanceCase:

```elixir
{:ok, session} = Wallaby.start_session(metadata: metadata)
session = Wallaby.Browser.resize_window(session, 300, 500)
{:ok, session: session}
```

That way each session would be sized correctly. We could probably support this as a configuration option. It probably makes sense to open to window larger then 100x100 by default.

**Comment by PragTob** [comment]
> It probably makes sense to open to window larger then 100x100 by default.

Most definitely. I imagine it being a gotcha moment in the first couple of tests that that's why maybe the menu doesn't show :D

**Comment by DavidVII** [comment]
Thanks a lot for that, @keathley. That clears things up quite a bit. I was wondering how to go about setting that in the `AcceptanceCase` file.

However, I'm wondering now weather or not a config is necessary. Perhaps adding a line or two about that in the README could suffice?

What do you guys think would be a good default? Perhaps something like 960x500? I'm happy to submit a PR over the next few days if you'd like.

**Comment by PragTob** [comment]
I think an addition to the README would be quite nice, and agree that it's an easy enough change that it doesn't need a config imo :)

**Comment by vanderhoop** [comment]
Just got nailed by this myself! I'm testing real-time updates across different sessions, so I'm consistently starting a second session. For now, I'm resizing the second session's window winthin my real-time update tests such that it matches the size I've given to the first session in my `IntegrationCase` setup, but @keathley, I'm happy to submit a PR for passing screensize options to `Wallaby.start_session` if you're down with it. 

Instead of 

```elixir
 {:ok, session_two} = Wallaby.start_session()
 session_two = resize_window(session_two, 960, 500)
```
It'd be something like:

```elixir
 {:ok, session_two} = Wallaby.start_session(
  [some_driver_initialization_config: "derp"],
  window_dimensions: [960, 500]
)
```
If you'd rather keep `Wallaby.start_session` limited to just starting the driver, I totally get it, but I have a feeling other folks are gonna hit this issue.

**Comment by keathley** [comment]
I'm OK with accepting something like `window_size` or `window_dimensions` as an option to `Wallaby.start_session/1`. If someone wants to submit a PR for that I'd be happy to take a look.

**Comment by zedtux** [comment]
```elixir
{:ok, session} = Wallaby.start_session(metadata: metadata)
session = Wallaby.Browser.resize_window(session, 300, 500)
{:ok, session: session}
```

is resizing the window in a strange way. I'm using bootstrap and I'm working on registering an account using a sign up button from the navbar, but my navbar is always blank because the CSS class `collapse` (having a media query) is hidding it.
Increasing the window size doesn't fix the issue, even with a width of `2000`.

To test it, I removed the CSS for the navbar responsiveness and used a width of `1500` and see the result:

![1513161074351045533](https://user-images.githubusercontent.com/478564/33934789-35f91552-dffa-11e7-99ed-a3d6f568060f.png)

As you can see, the window is well resized, but the content of the navbar is like it is in a very small window, while it should look like this:

![screenshot-2017-12-13 product hunt](https://user-images.githubusercontent.com/478564/33934909-7e61a46c-dffa-11e7-9774-75b0f1c83d07.png)

**Comment by keathley** [comment]
@zedtux which browser are you using?

**Comment by zedtux** [comment]
The second screenshot has been done with Firefox 57.0.1 and the first one is from Wallaby.

**Comment by keathley** [comment]
I'm wondering if this is a phantom bug. You might try with chrome to see what that does. Phantom has issues with new-ish css and such.

**Comment by mhanberg** [comment]
>I'm OK with accepting something like window_size or window_dimensions as an option to Wallaby.start_session/1. If someone wants to submit a PR for that I'd be happy to take a look.

I will take a crack at this.

**Comment by keathley** [comment]
@mhanberg were you able to find anything?

**Comment by mhanberg** [comment]
I have been exploring how the different drivers work and getting a feel for the codebase, but haven't yet written any code yet.

**Comment by mhanberg** [comment]
This can be closed due to #390
